src/Raven.Server/Documents/Indexes/Workers/TimeSeries/MapTimeSeries.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Client;
+using Raven.Server.Config.Categories;
+using Raven.Server.Documents.Indexes.MapReduce;
+using Raven.Server.Documents.TimeSeries;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Smuggler.Documents;
+
+namespace Raven.Server.Documents.Indexes.Workers.TimeSeries
+{
+    public sealed class MapTimeSeries : MapItems
+    {
+        private readonly TimeSeriesStorage _timeSeriesStorage;
+
+        public MapTimeSeries(Index index, TimeSeriesStorage timeSeriesStorage, IndexStorage indexStorage, MapReduceIndexingContext mapReduceContext, IndexingConfiguration configuration)
+            : base(index, indexStorage, mapReduceContext, configuration)
+        {
+            _timeSeriesStorage = timeSeriesStorage;
+        }
+
+        protected override IEnumerable<IndexItem> GetItemsEnumerator(DocumentsOperationContext databaseContext, string collection, long lastEtag, int pageSize)
+        {
+            foreach (var timeSeries in GetTimeSeriesEnumerator(databaseContext, collection, lastEtag, pageSize))
+            {
+                yield return new IndexItem(timeSeries.Key, timeSeries.Key, timeSeries.Etag, default, timeSeries.Name, timeSeries.SegmentSize, timeSeries);
```

Comment:
TODO:
check if putting key as Key and LoweredKey is problematic for TS or not

Reply:
https://github.com/ravendb/ravendb/pull/9972